File: src/main/java/com/refinedmods/refinedstorage/screen/CrafterTileDataParameterClientListener.java
package com.refinedmods.refinedstorage.screen;

import com.refinedmods.refinedstorage.screen.widget.sidebutton.CrafterModeSideButton;
import com.refinedmods.refinedstorage.tile.data.TileDataParameterClientListener;

public class CrafterTileDataParameterClientListener implements TileDataParameterClientListener<Boolean> {
    @Override
    public void onChanged(boolean initial, Boolean hasRoot) {
        if (!hasRoot) {
            BaseScreen.executeLater(CrafterScreen.class, gui -> gui.addSideButton(new CrafterModeSideButton(gui)));
        }
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/screen/CrafterTileDataParameterClientListener.kt
package com.refinedmods.refinedstorage.screen

import com.refinedmods.refinedstorage.screen.widget.sidebutton.CrafterModeSideButton
import com.refinedmods.refinedstorage.tile.data.TileDataParameterClientListener
import java.util.function.Consumer

class CrafterTileDataParameterClientListener : TileDataParameterClientListener<Boolean?> {
    override fun onChanged(initial: Boolean, hasRoot: Boolean) {
        if (!hasRoot) {
            BaseScreen.Companion.executeLater(CrafterScreen::class.java, Consumer { gui: CrafterScreen -> gui.addSideButton(CrafterModeSideButton(gui)) })
        }
    }
}